smart-contracts/assembly/contracts/NFT/__tests__/NFTEnumerable-example.spec.ts

```diff
@@ -0,0 +1,310 @@
+import {
+  changeCallStack,
+  resetStorage,
+  setDeployContext,
+} from '@massalabs/massa-as-sdk';
+
+import {
+  Args,
+  byteToBool,
+  bytesToString,
+  bytesToU256,
+} from '@massalabs/as-types';
+import { u256 } from 'as-bignum/assembly';
+import {
+  approve,
+  balanceOf,
+  burn,
+  constructor,
+  getApproved,
+  isApprovedForAll,
+  mint,
+  name,
+  ownerAddress,
+  ownerOf,
+  setApprovalForAll,
+  symbol,
+  totalSupply,
+  transferFrom,
+} from '../NFTEnumerable-example';
+
+import { getOwnedTokens } from './helpers';
+
+const NFTName = 'MASSA_NFT';
+const NFTSymbol = 'NFT';
+const contractOwner = 'A12UBnqTHDQALpocVBnkPNy7y5CndUJQTLutaVDDFgMJcq5kQiKq';
+const tokenAddress = 'AS12BqZEQ6sByhRLyEuf0YbQmcF2PsDdkNNG1akBJu9XcjZA1eT';
+const from = 'AU12CzoKEASaeBHnxGLnHDG2u73dLzWWfgvW6bc4L1UfMA5Uc5Fg7';
+const to = 'AU178qZCfaNXkz9tQiXJcVfAEnYGJ27UoNtFFJh3BiT8jTfY8P2D';
+const approved = 'AU1sF3HSa7fcBoE12bE1Eq2ohKqcRPBHuNRmdqAMfw8WEkHCU3aF';
+const zeroAddress = '';
+const tokenIds = [
+  u256.One,
+  u256.fromU32(2),
+  u256.fromU32(3),
+  u256.fromU32(4),
+  u256.fromU32(5),
+];
+
+function switchUser(user: string): void {
+  changeCallStack(user + ' , ' + tokenAddress);
+}
+
+beforeEach(() => {
+  resetStorage();
+  switchUser(contractOwner);
+  setDeployContext(contractOwner);
+  constructor(new Args().add(NFTName).add(NFTSymbol).serialize());
+});
+
+describe('NFT Enumerable Contract', () => {
+  describe('Initialization', () => {
+    test('should return correct name and symbol', () => {
+      expect(name()).toBe(NFTName);
+      expect(symbol()).toBe(NFTSymbol);
+    });
+    test('should return correct contract owner', () => {
+      expect(bytesToString(ownerAddress([]))).toBe(contractOwner);
+    });
+  });
+
+  describe('Minting', () => {
+    test('should mint a token to an address', () => {
+      mint(new Args().add(to).add(tokenIds[0]).serialize());
+      expect(
+        bytesToString(ownerOf(new Args().add(tokenIds[0]).serialize())),
+      ).toBe(to);
+      expect(
+        bytesToU256(balanceOf(new Args().add(to).serialize())),
+      ).toStrictEqual(u256.One);
+      expect(bytesToU256(totalSupply([]))).toStrictEqual(u256.One);
+    });
+
+    test('should mint multiple tokens to different addresses', () => {
+      mint(new Args().add(to).add(tokenIds[0]).serialize());
+      mint(new Args().add(from).add(tokenIds[1]).serialize());
+      expect(
+        bytesToU256(balanceOf(new Args().add(to).serialize())),
+      ).toStrictEqual(u256.One);
+      expect(
+        bytesToU256(balanceOf(new Args().add(from).serialize())),
+      ).toStrictEqual(u256.One);
+      expect(bytesToU256(totalSupply([]))).toStrictEqual(u256.fromU32(2));
+    });
+
+    test('should not mint to zero address', () => {
+      expect(() => {
+        mint(new Args().add(zeroAddress).add(tokenIds[0]).serialize());
+      }).toThrow('Unauthorized to');
+    });
```

Comment:
not sure this test make sense. maye a "test mint to an invalid address" would be more relevant

Reply:
thanks